### PR TITLE
Darkmoon Card: Decay + Sandstorm

### PR DESCRIPTION
**Pull request by tyler-cb**
https://www.wowhead.com/classic/item=221307/darkmoon-card-decay
https://www.wowhead.com/classic/item=221309/darkmoon-card-sandstorm

Placeholder proc rate of 2.5%, 5% spellpower scaling used for both, and leech amount of 30 for decay proc

### CODE REVIEW

**Comment by kayla-glick** [comment]
Is there precedent for a 2.5% proc rate? I'm just wary misleading users into thinking they're better than they actually are

**Comment by tyler-cb** [comment]
> Is there precedent for a 2.5% proc rate? I'm just wary of potentially misleading users into thinking they're better than they actually are

I based it on Darkmoon Card Maelstrom which appears to have a roughly 2-3% proc based on old discord posts and information online; lightning strike on here are DMC maelstrom procs:

2k

File: sim/common/sod/item_effects/phase_3.go [hunk 1]
```diff
@@ -89,6 +92,110 @@ func init() {
...
+				// Placeholder damage and coefficient values, update when P3 releases
+				result := spell.CalcAndDealDamage(sim, target, 30+0.05*spell.SpellDamage(), spell.OutcomeMagicHitAndCrit)
+				if result.Landed() {
+					spell.CalcAndDealHealing(sim, &character.Unit, result.Damage, spell.OutcomeHealing)
+					targetAura.Activate(sim)
+					targetAura.AddStack(sim)
+				}
+				if targetAura.GetStacks() == 5 {
+					// Placeholder coefficient
+					decayStackedSpell.CalcAndDealDamage(sim, target, sim.Roll(120, 180)+0.05*spell.SpellDamage(), spell.OutcomeMagicHitAndCrit)
```
Comment (kayla-glick) [inline sim/common/sod/item_effects/phase_3.go:140]: It looks like Darkmoon Card: Maelstrom doesn't have a SP coefficient listed https://www.wowhead.com/classic/spell=23687/lightning-strike. Do you know if it actually has one? I wonder if we should default to no coefficient for now

File: sim/common/sod/item_effects/phase_3.go [hunk 2]
```diff
@@ -89,6 +92,110 @@ func init() {
...
+			SpellSchool: core.SpellSchoolNature,
+			DefenseType: core.DefenseTypeMagic,
+			ProcMask:    core.ProcMaskEmpty,
+
+			DamageMultiplier: 1,
+			ThreatMultiplier: 1,
+
+			ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+				for _, aoeTarget := range sim.Encounter.TargetUnits {
+					spell.CalcAndDealDamage(sim, aoeTarget, sim.Roll(200, 300)+0.05*spell.SpellDamage(), spell.OutcomeMagicHitAndCrit)
```
Comment (kayla-glick) [inline sim/common/sod/item_effects/phase_3.go:177]: Same comment about the SP coefficient from above

**Pull request merged**